wandb/sync/sync.py

```diff
@@ -273,7 +297,8 @@ def run(self):
                         sys.stdout.flush()
                         shown = True
             sm.finish()
-            if self._mark_synced and not self._view:
+            # Only mark synced if the run actually finished
+            if self._mark_synced and not self._view and finished:
```

Comment:
The other big change is we don't mark "live" runs as synced.

Reply:
good.